src/EditorFeatures/Test/Collections/IntervalTreeTests.cs

```diff
@@ -307,6 +307,70 @@ public void TestEmptySpanAtStart()
             Assert.Equal(1, tree.Count());
         }
 
+        private class Int32Introspector: IIntervalIntrospector<int>
+        {
+            public static Int32Introspector Instance = new Int32Introspector();
+
+            public int GetLength(int value)
+            {
+                return 0;
+            }
+
+            public int GetStart(int value)
+            {
+                return value;
+            }
+        }
+
+        private IntervalTree<int> CreateIntTree(params int[] values)
+        {
+            return new IntervalTree<int>(Int32Introspector.Instance, values);
+        }
+
+        [Fact]
+        public void TestSortedEnumerable1()
+        {
+            var tree = new IntervalTree<int>(Int32Introspector.Instance, new[] { 0, 0, 0 });
+
+            Assert.Equal(CreateIntTree(0, 0, 0), new[] { 0, 0, 0 });
```

Comment:
I could also do this with some sort of loop construct.  I'm not sure what you guys woudl prefer.  The loop construct would work a lot better if, say, we wanted to up the number of elements.